src/Type/Schema.php

```diff
@@ -359,12 +368,24 @@ private function loadType(string $typeName) : ?Type
 
     private function defaultTypeLoader(string $typeName) : ?Type
     {
-        // Default type loader simply fallbacks to collecting all types
+        // Default type loader simply falls back to collecting all types
         $typeMap = $this->getTypeMap();
 
         return $typeMap[$typeName] ?? null;
     }
 
+    /**
+     * @param Type|callable():Type $type
+     */
+    public static function resolveType($type) : Type
```

Comment:
I think this function can be removed, as this check and conversion happens above. The crafted error is nicer than having this fail at the function boundary.

Reply:
Ah, just saw this is used in a bunch of places. I am fine with keeping it, but we should avoid double-checking for `is_callable()` in this class.

Reply:
Why would I remove this? It provides a hook to narrow the type, encapsulates the actual work, and in any case is currently called 27 times in the code base. 

I'm doing my best to hang in there, but this PR has been open since October, and I am rapidly reaching the point where my main application can't progress without this feature. Any chance we can save major refactoring suggestions like this for some other day?

Reply:
I was wrong with my first comment, this function should stay. There is redundant code in this class that should go instead, see https://github.com/webonyx/graphql-php/pull/557#discussion_r439626066